File: yht0827/1월/1월 25일/추억 점수.java
import java.util.*;

class Solution {
	public int[] solution(String[] name, int[] yearning, String[][] photo) {
		int[] answer = new int[photo.length];

		for (int i = 0; i < photo.length; i++) {
			for (int j = 0; j < photo[i].length; j++) {

				if (Arrays.asList(name).indexOf(photo[i][j]) != -1) {
					answer[i] += yearning[Arrays.asList(name).indexOf(photo[i][j])];
				}
			}
		}
		return answer;
	}
}

File: zuzubibi/old/1월/1월 25일/추억점수.py
def solution(name, yearning, photo):
    return [ sum(yearning[name.index(j)] for j in i if j in name) for i in photo]

# [19, 15, 6]
print(solution(["may", "kein", "kain", "radi"],[5, 10, 1, 3],[["may", "kein", "kain", "radi"],["may", "kein", "brin", "deny"], ["kon", "kain", "may", "coni"]]))
# [67, 0, 55]
print(solution(["kali", "mari", "don"],[11, 1, 55], [["kali", "mari", "don"], ["pony", "tom", "teddy"], ["con", "mona", "don"]]))
# [5, 15, 0]
print(solution(["may", "kein", "kain", "radi"], [5, 10, 1, 3], [["may"],["kein", "deny", "may"], ["kon", "coni"]]))